test_cmd.py

```diff
@@ -80,13 +80,20 @@ def read_file(file_name):
 
     def run_cmd(self, input_text):
         global max_parallel
+        stdout = None
+        stderr = None
+        process = None
+
         max_parallel.acquire()
+        try:
+
+            process = Popen(self.cmd, stdout=PIPE, stdin=PIPE, stderr=PIPE)
+            stdout, stderr = process.communicate(input=input_text)
+        finally:
+            max_parallel.release()
 
-        process = Popen(self.cmd, stdout=PIPE, stdin=PIPE, stderr=PIPE)
-        stdout, stderr = process.communicate(input=input_text)
+        return stdout, stderr, process
```

Comment:
I see you're returning `process` here! But I don't see the returned `process` being used (in the `detail` function). Is this for future use? :slightly_smiling_face: